sympy/integrals/risch.py

```diff
@@ -761,6 +764,29 @@ def decrement_level(self):
         self.case = self.cases[self.level]
         return None
 
+
+class UnevaluatedTan(Function):
+    """
+    Used to avoid auto-simplification of tangents of the type
+    'tan(I*x)'.
```

Comment:
There is no need to avoid this kind of auto-simplification. When there is an `I` in the expression, real type monomial extensions (`tan`, `atan`) should not be used. In this case, `tanh(x)` is a rational expression of `exp(x)`, so an `exp` monomial can be used to compute the integral.

Reply:
But this needs to be done or else you will get incorrect results, e.g., using `tan(x).rewrite(exp) + tan(x)`.  

I don't know if we should be worried about other forms of complex constants (like `root(-1, 3)`). I guess if a constant is numeric we can always test if it is nonreal, and if it is symbolic, we can implicitly assume that it is real.

Reply:
> tan(x).rewrite(exp) + tan(x)

I don't see how this could lead to incorrect results. If `tan(x)` was rewritten in (a part of) the input expression, then that would contain `I` and  `rewrite_complex` would be True. And then all the rest of the trigonometric functions would also be rewritten.

The implicit complex constants should be taken into account at some point, but I think that `I` could suffice for now. That is obviously the most common one.

Reply:
Sorry to clarify, I mean the `I` rewriting needs to be done or you will get wrong results. (right now it's in a separate PR)